pkg/operation/seed/seed.go

```diff
@@ -350,6 +350,16 @@ func RunReconcileSeedFlow(
 		if err := common.DeleteHvpa(ctx, k8sSeedClient, v1beta1constants.GardenNamespace); client.IgnoreNotFound(err) != nil {
 			return err
 		}
+	} else {
+		// Clean up stale vpa objects
+		resources := []client.Object{
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+			&autoscalingv1beta2.VerticalPodAutoscaler{ObjectMeta: metav1.ObjectMeta{Name: "aggregate-prometheus-vpa", Namespace: v1beta1constants.GardenNamespace}},
+		}
+
+		if err := kutil.DeleteObjects(ctx, k8sSeedClient.Client(), resources...); client.IgnoreNotFound(err) != nil {
```

Comment:
There is no need for `client.IgnoreNotFound(err) != nil` as this check is already done in `DeleteObject` func. See https://github.com/gardener/gardener/blob/779eb21a6118f81d1e0cb2fd92463158cdc643f6/pkg/utils/kubernetes/object.go#L47

```suggestion
		if err := kutil.DeleteObjects(ctx, k8sSeedClient.Client(), resources...); err != nil {
```